vms.tf

```
@@ -25,7 +25,7 @@ resource "proxmox_virtual_environment_vm" "this" {
     )
   }
 
-  name      = format("%s-%s", each.value.type, random_string.this[each.key].result)
+  name      = format("%s-%s-%s", var.talos.node_data.node_prefix ,each.value.type, random_string.this[each.key].result)
```

```suggestion
  name      = format("%s%s-%s", var.talos.node_data.node_prefix ,each.value.type, random_string.this[each.key].result)
```
I think the hyphen should just be set by the user in the variable as it's optional won't effect existing names.